src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -2345,7 +2398,113 @@ public static void copyImplementation(Design src, Design dest, boolean lockPlace
 					p.setIsPIPFixed(true);
 				}
 			}
-		}		
+		}
+
+		if (copyStaticNets) {
+			copyStaticNets(dest, staticNets, siteInstsOfCells);
+		}
+	}
+
+	/**
+	 * Copy the route of static nets feeding the sinks within the given SiteInst.
+	 * @param dest The destination design
+	 * @param staticNets The list of static nets to copy
+	 * @param siteInstsOfCells The set of SiteInst containing the sinks of the static nets
+	 */
+	private static void copyStaticNets(Design dest, List<Net> staticNets, Set<SiteInst> siteInstsOfCells) {
+		// This method traces a route similar to that in getTrimmablePIPsFromPins. However, there is one subtle difference.
+		// Let's consider a route from one GND source (S) to two sinks (T1 and T2) and the last common node/pip is X.
+		// If only T1 is in a cell to be copied, the tracing code here will extract all pips from S to T1.
+		// However, getTrimmablePIPsFromPins return only the pips from X to T1.
+
+		// Map from a node to its driver PIP
+		// Note: Some PIPs are bidirectional. But, every PIP performs only one direction.
+		// The direction of a bidirectional PIP is determined from the context, ie., its connecting directional PIPs.
+		// To determine that context, go through directional PIPs first. This process does not support consecutive bidirectional PIPs.
+		Map<Net,Map<Node,PIP>> netToUphillPIPMap = new HashMap<>();
+		// netToPIPs to store PIPs extracted for static nets.
+		Map<Net,Set<PIP>> netToPIPs = new HashMap<>();
+		for (Net net : staticNets) {
+			netToPIPs.put(net, new HashSet<>());
+			Map<Node,PIP> nodeToDriverPIP = new HashMap<>();
+			List<PIP> biPIPs = new ArrayList<>();
+			// TODO: This goes through the whole design. Try to limit it to the target cell, if possible.
```

Comment:
Is this TODO still relevant?

Reply:
Yes. I don't know how to do that and don't even know if it is possible because static nets can go through many hierarchy.  

If there is no way to do it, should it be removed?

Reply:
Take out TODO. Mention limitation in javadoc.